src/Sylius/Bundle/ApiBundle/CommandHandler/RegisterShopUserHandler.php

```diff
@@ -32,14 +36,24 @@ final class RegisterShopUserHandler implements MessageHandlerInterface
     /** @var CustomerProviderInterface */
     private $customerProvider;
 
+    /** @var CartToUserAssignerInterface */
+    private $cartToUserAssignerInterface;
+
+    /** @var bool */
+    private $assignCartToRegisterUser;
```

Comment:
This is just a proposal, but with this shop owner can change if he want registering user to has cart assigned or not with just parameter in services.xml 
I have made same to login in user.
Wdyt?

Reply:
IMHO it should be only one logic: after registration user has the previous cart. if someone wants to have a different 
cart flow, he should make customization with it.

Reply:
Removed 😃